### PR TITLE
[SPARK-26725][TEST] Fix the input values of UnifiedMemoryManager constructor in test suites

### PR DESCRIPTION
**Pull request by srowen**
## What changes were proposed in this pull request?

Adjust mem settings in UnifiedMemoryManager used in test suites to ha…ve execution memory > 0
Ref: https://github.com/apache/spark/pull/23457#issuecomment-457409976

## How was this patch tested?

Existing tests

### CODE REVIEW

**Comment by gatorsmile** [comment]
It sounds like there are more : https://github.com/apache/spark/pull/23457/files#r250891226

It impacts not only the test cases but also the actual code.

**Comment by gatorsmile** [comment]
Also cc @cloud-fan @zsxwing Could you take a look?

**Comment by srowen** [comment]
Oops yes I should have marked this as WIP, just using Jenkins to test part of the change. Yes it occurred in more places than just the tests, this is important.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala [hunk 1]
```diff
@@ -97,7 +97,7 @@ private[execution] object HashedRelation {
         new UnifiedMemoryManager(
           new SparkConf().set(MEMORY_OFFHEAP_ENABLED.key, "false"),
           Long.MaxValue,
-          Long.MaxValue,
+          Long.MaxValue / 2,
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala]: shall we add an assert in `UnifiedMemoryManager` to make sure execution memory is > 0?

Comment (srowen) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala]: That's fine by me. I presume it's not strictly an error to specify 0 execution memory, but also can't think of a situation where it's valid.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala [hunk 2]
```diff
@@ -230,7 +230,7 @@ private[joins] class UnsafeHashedRelation(
       new UnifiedMemoryManager(
         new SparkConf().set(MEMORY_OFFHEAP_ENABLED.key, "false"),
         Long.MaxValue,
-        Long.MaxValue,
+        Long.MaxValue / 2,
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala]: note: This actually doesn't matter. It's a dummy memory manager which is only used to create the `BytesToBytesMap`. This memory manager will not consume any storage memory, which means the execution memory can grow until it takes all the memory, which is `Long.MaxValue`.

Anyway, it's still better to make sure the values are not nonsense.

File: core/src/main/scala/org/apache/spark/memory/MemoryManager.scala [hunk 3]
```diff
@@ -41,6 +41,9 @@ private[spark] abstract class MemoryManager(
     onHeapStorageMemory: Long,
     onHeapExecutionMemory: Long) extends Logging {
 
+  require(onHeapStorageMemory > 0, "onHeapStorageMemory must be > 0")
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/memory/MemoryManager.scala]: Hm, on second thought, there's not really a reason this must be positive. It's at least possible that no memory is immune from eviction.

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**